### PR TITLE
MTL+ add encode vp9 10bit non-lowpower test path support

### PR DESCRIPTION
**Pull request by FocusLuo**
1, ffmpeg/gstreamer test add vp9-10bit non-lowpower mode support
2, ffmpeg/gstreamer lib update check for vp9-10bit non-lowpower
3, MTL caps

### CODE REVIEW

**Review by uartie — CHANGES_REQUESTED** [review]
We can do better... this is too hacked and confusing

File: lib/ffmpeg/encoderbase.py [hunk 1]
```diff
@@ -141,6 +141,16 @@ def before(self):
...
 
+  def get_lowpower_opt(self):
+    if get_media()._get_gpu_gen() < 13:
+         #only cover "low_power=0" path
+         return 0
+    else:
+         #For MTL platforms (gen >== 13)
+         #will add random() to set "low_power=0" and no "low_power" option
+         #to cover more test path in future
+         return 0
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply lib/ffmpeg/encoderbase.py]: updated, thanks

File: lib/ffmpeg/vaapi/encoder.py [hunk 2]
```diff
@@ -99,7 +105,7 @@ def check_output(self):
     ipbmode = 0 if vars(self).get("gop", 0) <= 1 else 1 if vars(self).get("bframes", 0) < 1 else 2
     ipbmsgs = [
       "Using intra frames only",
-      "Using intra and P-frames|[L|l]ow delay|forward-prediction"
+      "|Using intra and P-frames|[L|l]ow delay|forward-prediction"
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply lib/ffmpeg/vaapi/encoder.py]: updated, thanks review

File: test/gst-vaapi/encode/10bit/vp9.py [hunk 3]
```diff
@@ -28,6 +28,18 @@ def before(self):
   def get_file_ext(self):
     return "webm"
 
+@slash.requires(*platform.have_caps("encode", "vp9_10"))
+class VP9_10EncoderTest(VP9_10EncoderBaseTest):
+  def before(self):
+    super().before()
+    _lowpower_ = self.get_lowpower_opt()
+    if 0 == _lowpower_:
+      _lowpower_ = False
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply test/gst-vaapi/encode/10bit/vp9.py]: updated, thanks review

**Comment by FocusLuo** [comment]
> We can do better... this is too hacked and confusing

updated, thanks review

**Comment by FocusLuo** [comment]
user@P12SL03SLIM50:/tmp_sda4/work_focus/applications.media.middleware.validation.vaapi-fits-smoke$ ./smoke run test/gst-va/encode/10bit/vp9.py --pl MTL --context=sanity -vv; ./smoke run test/gst-msdk/encode/10bit/vp9.py --pl MTL --context=sanity -vv; ./smoke run test/ffmpeg-vaapi/encode/10bit/vp9.py --pl MTL --context=sanity -vv; ./smoke run test/ffmpeg-qsv/encode/10bit/vp9.py --pl MTL --context=sanity -vv;
all passed

**Comment by uartie** [comment]
please check test failures

File: test/ffmpeg-qsv/encode/10bit/vp9.py [hunk 4]
```diff
@@ -6,10 +6,68 @@
...
+    vars(self).update(
+      case      = case,
+      gop       = 30 if ipmode != 0 else 1,
+      qp        = qp,
+      slices    = slices,
+      rcmode    = "cqp",
+      quality   = quality,
+    )
+
+  @parametrize_with_unused(*gen_vp9_cqp_lp_parameters(spec), ['refmode', 'loopshp', 'looplvl'])
```
Comment (uartie) [inline test/ffmpeg-qsv/encode/10bit/vp9.py:25]: wrong case variant generators.

**Review by uartie — CHANGES_REQUESTED** [review]

File: lib/ffmpeg/vaapi/encoder.py [hunk 5]
```diff
@@ -73,9 +73,15 @@ def check_output(self):
     assert m is not None, "Possible incorrect profile used"
 
     # entrypoint
+    if "jpeg" == self.codec:
+      _VAEntrypoint_Encode_Mode_ = "VAEntrypointEncPicture"
+    elif self.codec in ["vp9", "vp9-10"]:
+      _VAEntrypoint_Encode_Mode_ = "VAEntrypointEncSliceLP"
+    else :
+      _VAEntrypoint_Encode_Mode_ = "VAEntrypointEncSlice"
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply lib/ffmpeg/vaapi/encoder.py]: updated

File: lib/gstreamer/encoderbase.py [hunk 6]
```diff
@@ -87,6 +87,16 @@ def map_profile(self):
...
 
+  def get_lowpower_opt(self):
+    if get_media()._get_gpu_gen() < 13:
+         #only cover "low_power=0" path
+         return 0
+    else:
+         #For MTL platforms (gen >== 13)
+         #will add random() to set "low_power=0" and no "low_power" option
+         #to cover more test path in future
+         return 0
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply lib/gstreamer/encoderbase.py]: updated

**Comment by FocusLuo** [comment]
> please check test failures

depend on https://github.com/intel/vaapi-fits/pull/565  (lib/parameters.py file)

File: test/ffmpeg-qsv/encode/10bit/vp9.py (hunk 4 above)
Comment (FocusLuo) [reply test/ffmpeg-qsv/encode/10bit/vp9.py]: updated

**Comment by uartie** [comment]
CI pre-check autogenerated 10-bit VP9 CQP cases and it looks like they all fail on i965 driver.  Thus, they will fail in sanity/full, too.  I see the i965 caps for KBL/CFL both include VME VP9 10bit... does i965 "actually" support it?

Also, please add case definitions to our BAT (pre-check) suite, otherwise >200 CQP cases will get generated automatically for each component.

**Comment by uartie** [comment]
>  I see the i965 caps for KBL/CFL both include VME VP9 10bit... does i965 "actually" support it?

Nope... https://github.com/intel/vaapi-fits/pull/566

**Comment by uartie** [comment]
Awaiting PR for BAT (pre-check).

**Comment by uartie** [comment]
> Awaiting PR for BAT (pre-check).

https://github.com/intel-innersource/applications.media.middleware.validation.vaapi-fits-smoke/pull/354

**Pull request merged**